error from `loadPackage` swallowed in node 0.10 and 0.12

While debugging why Travis CI was running into failing builds only with Node.js 4.2 for my ESLint plugin, I've noticed the errors from lib/config/config-file.js's `loadPackage` function are swallowed in Node.js 0.10 and 0.12.

[cloudflare/eslint-plugin-cflint Travis CI run](https://travis-ci.org/cloudflare/eslint-plugin-cflint/jobs/96641651)

In Node.js 4.2 the following error is reported with a non-zero exit code:

```
Error: Cannot read config package: eslint-config-xo
Error: Cannot find module 'eslint-config-xo'
Referenced from: 
    at Function.Module._resolveFilename (module.js:337:15)
    at Function.Module._load (module.js:287:25)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at loadPackage (/home/travis/build/cloudflare/eslint-plugin-cflint/node_modules/eslint/lib/config/config-file.js:168:16)
    at loadConfigFile (/home/travis/build/cloudflare/eslint-plugin-cflint/node_modules/eslint/lib/config/config-file.js:212:18)
    at load (/home/travis/build/cloudflare/eslint-plugin-cflint/node_modules/eslint/lib/config/config-file.js:385:18)
    at /home/travis/build/cloudflare/eslint-plugin-cflint/node_modules/eslint/lib/config/config-file.js:326:36
    at Array.reduceRight (native)
    at Object.applyExtends (/home/travis/build/cloudflare/eslint-plugin-cflint/node_modules/eslint/lib/config/config-file.js:309:28)
```

Despite having the same tree, 0.10 and 0.12 do not print a warning and exit successfully. **This is a lie**.

If you run with the `DEBUG` environment variable set, you see that the debug mess from line 170 runs.

```
  eslint:config-file Loading xo +0ms
  eslint:config-file Loading config package: eslint-config-xo +1ms
  eslint:config-file Error reading package: eslint-config-xo +5ms
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/29095576-error-from-loadpackage-swallowed-in-node-0-10-and-0-12?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/bin/eslint.js
+++ b/bin/eslint.js
@@ -60,10 +60,16 @@ if (useStdIn) {
     exitCode = cli.execute(process.argv);
 }
 
-/*
- * Wait for the stdout buffer to drain.
- * See https://github.com/eslint/eslint/issues/317
- */
-process.on("exit", function() {
-    process.exit(exitCode);
-});
+// https://github.com/eslint/eslint/issues/4691
+// In Node.js >= 0.12, you can use a cleaner way
+if ("exitCode" in process) {
+    process.exitCode = exitCode;
+} else {
+    /*
+     * Wait for the stdout buffer to drain.
+     * See https://github.com/eslint/eslint/issues/317
+     */
+    process.on("exit", function() {
+        process.exit(exitCode);
+    });
+}

